lib/rubocop/cop/primer/primer_octicon.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+require "rubocop"
+require "primer/classify/utilities"
+
+# :nocov:
+module RuboCop
+  module Cop
+    module Primer
+      # This cop ensures that components use System Arguments instead of CSS classes.
+      #
+      # bad
+      # octicon("icon")
+      #
+      # good
+      # primer_octicon("icon")
+      class PrimerOcticon < RuboCop::Cop::Cop
+        INVALID_MESSAGE = <<~STR
+          Replace the octicon helper with primer_octicon. See https://primer.style/view-components/components/octicon for details.
+        STR
+
+        SIZE_ATTRIBUTES = %i[height width size].freeze
+        VALID_ATTRIBUTES = [*SIZE_ATTRIBUTES, :class].freeze
+        INVALID_ATTRIBUTE = -1
+
+        def on_send(node)
+          return unless node.method_name == :octicon
+          return unless node.arguments?
+
+          kwargs = kwargs(node)
+          attributes = kwargs.keys.map(&:value)
+
+          # Don't convert unknown attributes
+          return unless (attributes - VALID_ATTRIBUTES).empty?
+          # Can't convert size
+          return if octicon_size_attributes(kwargs) == INVALID_ATTRIBUTE
+
+          # find class pair
+          classes = classes(kwargs)
+
+          return if classes == INVALID_ATTRIBUTE
+
+          # check if classes are convertible
+          if classes.present?
+            system_arguments = ::Primer::Classify::Utilities.classes_to_hash(classes)
+
+            # Uses custom classes
+            return if system_arguments[:classes].present?
+          end
+
+          add_offense(node, message: INVALID_MESSAGE)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            icon = node.arguments.first.source
+            kwargs = node.arguments.last
+
+            # Converting arguments for the component
+            classes = classes(kwargs)
+            size_attributes = transform_sizes(kwargs)
+            args = arguments_as_string(icon, size_attributes, classes)
+
+            corrector.replace(node.loc.expression, "primer_octicon(#{args})")
+          end
+        end
+
+        private
+
+        def transform_sizes(kwargs)
+          attributes = octicon_size_attributes(kwargs)
+
+          attributes.transform_values do |size|
+            if size.between?(10, 16)
+              ""
+            elsif size.between?(22, 26)
+              ":medium"
```

Comment:
Are we sure about this mapping rule? I'm not so sure about automating this kind of change without visually verifying the results.

Reply:
This is something I was doing on the dotcom side

Reply:
@jonrohan if you're OK with this change, I am!